src/test/java/PlayAgainMenuTest.java

```diff
@@ -0,0 +1,47 @@
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class PlayAgainMenuTest {
+    @Test
+    public void itWillDisplayTheMenu() {
+        FakeUI ui = new FakeUI();
+        ui.input.add("Y");
+        ui.input.add("N");
+
+        new PlayAgainMenu(ui).display();
+
+        assertEquals("Play again?\nY/N:  Do you want to use a different game configuration?\nY/N:  ", ui.log);
+    }
+
+    @Test
+    public void itWillOnlyAcceptAcceptYOrN() {
+        FakeUI ui = new FakeUI();
+        ui.input.add("asdk");
+        ui.input.add("0091");
+        ui.input.add("N");
```

Comment:
This test seems to be testing a few things, including the implementation of your `FakeUI` class (the top-level one), where it has that `while(!validator.isValid(userInput))` loop. I'm guessing that loop appears somewhere in the production code (outside of `FakeUI`), so I think either that code should be used by `FakeUI`, or you can find some other way to avoid that looping concept from being duplicated (and hence being error-prone) across tests & production code.